From a7f0b70df695e16c06fa92cb4eaa9c9129c5304f Mon Sep 17 00:00:00 2001
From: Harshesh Valera <harshesh.v@variscite.com>
Date: Fri, 4 Jun 2021 21:12:09 -0700
Subject: [PATCH] imx8mn_var_som: Use property to override fw name

- For imx8mn_var_som 2Ghz som uses different firmware

Signed-off-by: Harshesh Valera <harshesh.v@variscite.com>
---
 src/bt_vendor_brcm.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/bt_vendor_brcm.c b/src/bt_vendor_brcm.c
index 3fcdbd4..8f4eeeb 100644
--- a/src/bt_vendor_brcm.c
+++ b/src/bt_vendor_brcm.c
@@ -106,6 +106,7 @@ static int init(const bt_vendor_callbacks_t* p_cb, unsigned char *local_bdaddr)
     char fw_path_value[PROPERTY_VALUE_MAX];
     char fw_id_value[PROPERTY_VALUE_MAX];
     char fw_file_value[PROPERTY_VALUE_MAX];
+    char imx8mn_value[PROPERTY_VALUE_MAX];
     int  fd, sz;
     char buf[CONF_MAX_LINE_LEN];
 
@@ -165,6 +166,16 @@ static int init(const bt_vendor_callbacks_t* p_cb, unsigned char *local_bdaddr)
            }
        }
     }
+    /* check for som_imx8mn present */
+    property_get("ro.boot.som_imx8mn", imx8mn_value, "0");
+    if ((strcmp(imx8mn_value, "0") != 0))
+    {
+               property_get("ro.boot.bt_firmware", fw_file_value, "0");
+               if (strcmp(fw_file_value, "0") != 0)
+               {
+                   hw_set_patch_file_name(NULL, fw_file_value, 0);
+               }
+    }
     /* store reference to user callbacks */
     bt_vendor_cbacks = (bt_vendor_callbacks_t *) p_cb;
 
-- 
2.25.1

